src/main/java/it/pagopa/pagopa/apiconfig/service/IcaService.java

```diff
@@ -150,23 +152,63 @@ public XSDValidation verifyXSD(MultipartFile xml) {
 
     @Transactional
     public void createIca(@NotNull MultipartFile file, Boolean force) {
-        List<CheckItem> checks = verifyIca(file, force);
+        try{
+            createIca(new ByteArrayInputStream(file.getInputStream().readAllBytes()), force, false);
+        }catch(SAXException | IOException e){
+            throw new AppException(HttpStatus.BAD_REQUEST, ICA_BAD_REQUEST, "Problem when creating new ICA");
+        }
+    }
+
+    @Transactional
+    @java.lang.SuppressWarnings({"javasecurity:S6096", "java:S5443"})
+    public void createMassiveIcas(@NotNull MultipartFile file, Boolean force) {
+        try{
+            ZipInputStream zis = new ZipInputStream(file.getInputStream());
```

Comment:
could these code lines be generalized considering massive verification icas too?

Reply:
I'm trying, problem is two different functions (create and verify) are called in the two methods inside the same condition